lib/ddtrace/span.rb

```diff
@@ -254,6 +256,64 @@ def to_hash
       h
     end
 
+    # MessagePack serializer interface. Making this object
+    # respond to `#to_msgpack` allows it to be automatically
+    # serialized by MessagePack.
+    #
+    # This is more efficient than doing +MessagePack.pack(span.to_hash)+
+    # as we don't have to create an intermediate Hash.
+    #
+    # @param packer [MessagePack::Packer] serialization buffer, can be +nil+ with JRuby
+    def to_msgpack(packer = nil)
+      # As of 1.3.3, JRuby implementation doesn't pass an existing packer
+      packer ||= MessagePack::Packer.new
+
+      if !@start_time.nil? && !@end_time.nil?
+        packer.write_map_header(13) # Set header with how many elements in the map
+
+        packer.write('start')
+        packer.write((@start_time.to_f * 1e9).to_i)
```

Comment:
Since this is written in two places now, read only property?

`#start_time_ns` + `#duration_ns` ?

Reply:
Cool, updated it.
I also changed the operation to perform fewer operations to get the value as nanoseconds.
The move to a separate method, combined with the arithmetic improvements increased performance bit a very tiny bit.